src/xcode/ENA/ENA/Source/Services/TraceWarningDownload/TraceWarningPackageDownload.swift

```diff
@@ -275,6 +273,14 @@ class TraceWarningPackageDownload: TraceWarningPackageDownloading {
 				
 		// 7. Determine packagesToDownload
 		let packagesToDownload = determinePackagesToDownload(availables: availablePackagesOnCDN, earliest: earliestRelevantPackage)
+		
+		guard !packagesToDownload.isEmpty else {
```

Comment:
Is this really necessary for the bugfix or does this handle another bug?
Isn't that handled by the forEach in line 294 ff.? Because when the packages are empty, the success should be triggered anyway?
If not, I recommend to add a new case for `TraceWarningSuccess`, because this is not equal to the regular success.

Reply:
That is correct but the reason this guard is here is to prevent changing the status to "downloading". Also, there is no need to try to attempt to download an empty package array.

Reply:
When its preventing the status to change, it' fine. Otherwise i agree with you with the "no need", but actually its exactly implemented like the tech spec defines it. So I would be careful with any logical change. But it is okay for me here.